### PR TITLE
Update documentation for 0.23

### PR DESCRIPTION
**Pull request by nblock**
The goal is to update the documentation for the upcoming 0.23 release with minimal changes and (mostly) add one commit per file.

Other issues such as duplicate instructions in multiple files are not addressed as those require some discussions with the maintainers beforehand.

- [x] have read the [CONTRIBUTING.md](./CONTRIBUTING.md) file
- [ ] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [x] updated documentation if needed
- [ ] updated CHANGELOG.md

Fixes: #2090
Fixes: #1822

## Summary by CodeRabbit

- **Documentation**
	- Updated ACL configuration documentation to reflect a change in the parameter name from `acl_policy_path` to `policy.path`, enhancing clarity for users.
	- Revised exit node documentation to replace "Machine" with "Node" for improved consistency and readability, and simplified a URL by removing unnecessary query parameters.
	- Enhanced formatting and readability of shell command examples in the remote CLI documentation.
	- Made minor textual modifications in the reverse proxy documentation for clarity and grammatical accuracy.
	- Clarified instructions for running the Headscale Docker container, including updates to command references and placeholder formats.
	- Improved clarity and presentation in the Linux running documentation, including a more prominent warning format.
	- Corrected typographical errors and improved clarity in the OpenBSD running documentation.
	- Updated Sealos documentation to remove specific version references, promoting access to the latest templates.

### CODE REVIEW

**Comment by nblock** [comment]
As per https://github.com/juanfont/headscale/commit/1193a50e9ed260324de76e23a5744f6473ca0386, the oldest support Tailscale client version is 1.42. The documentation and the OS specific endpoints `/apple` and `/windows` distinguish between various versions; all of them older than 1.42.

Can I just get rid of this and assume 1.42+? This would remove some probably useless and no longer required instructions in the documentation and also from the OS specific endpoints `/apple` and `/windows`.

**Comment by kradalby** [comment]
> As per [1193a50](https://github.com/juanfont/headscale/commit/1193a50e9ed260324de76e23a5744f6473ca0386), the oldest support Tailscale client version is 1.42. The documentation and the OS specific endpoints `/apple` and `/windows` distinguish between various versions; all of them older than 1.42.
> 
> Can I just get rid of this and assume 1.42+? This would remove some probably useless and no longer required instructions in the documentation and also from the OS specific endpoints `/apple` and `/windows`.

Yes please!

Quick look at `/apple`, I think all the mentions of versions can be removed, I think that the profile and `defaults` stuff can still stay around as it does work still, and to some this might be preferred to force it to always try headscale first, but restructuring make sense.

Same for `/windows`, the registry stuff is useful, but dont need to distinguish the versions anymore, if you are up for it that would be great.

Preferably do it as a followup PR.

File: docs/running-headscale-container.md [hunk 1]
```diff
@@ -109,7 +102,7 @@ not work with alternatives like [Podman](https://podman.io). The Docker image ca
 
     ```shell
     docker exec headscale \
-    headscale users create myfirstuser
+      headscale users create myfirstuser
```
Comment (kradalby) [inline docs/running-headscale-container.md:105]: need to remove `headscale` here right?

```suggestion
      users create myfirstuser
```

File: docs/running-headscale-container.md [hunk 2]
```diff
@@ -124,7 +117,7 @@ To register a machine when running `headscale` in a container, take the headscal
 
 ```shell
 docker exec headscale \
-  headscale --user myfirstuser nodes register --key <YOU_+MACHINE_KEY>
+  headscale --user myfirstuser nodes register --key <YOUR_MACHINE_KEY>
```
Comment (kradalby) [inline docs/running-headscale-container.md:120]:
```suggestion
  --user myfirstuser nodes register --key <YOUR_MACHINE_KEY>
```

**Review by kradalby — APPROVED** [review]
Thank you!, these changes are incredibly helpful!

Two small changes, I might have confused myself gotten it the wrong way around with docker, but right now the serve and user example is not calling docker the same way.

**Comment by nblock** [comment]
> Preferably do it as a followup PR.

Thx for the feedback; will submit a new PR for this later.

File: docs/running-headscale-container.md (hunk 1 above)
Comment (nblock) [reply docs/running-headscale-container.md:105]: For `exec` the command is actually `headscale ...`, I tested the commands with docker on Debian 12.

File: docs/running-headscale-container.md (hunk 2 above)
Comment (nblock) [reply docs/running-headscale-container.md:120]: For `exec` the command is actually `headscale ...`, I tested the commands with docker on Debian 12.

**Pull request merged**